salt/modules/mount.py

```diff
@@ -1422,3 +1569,287 @@ def delete_mount_cache(real_name):
+        # NOTE: missing some special fstypes here
+        specialFSes = frozenset([
+            'none',
+            'tmpfs',
+            'sysfs',
+            'proc',
+            'fusectl',
+            'debugfs',
+            'securityfs',
+            'devtmpfs',
+            'cgroup',
+            'btrfs',
+            'cdrfs',
+            'procfs',
+            'jfs',
+            'jfs2',
+            'nfs',
+            'sfs',
+            'nfs3',
+            'cachefs',
+            'udfs',
+            'cifs',
+            'namefs',
+            'pmemfs',
+            'ahafs',
+            'nfs4',
+            'autofs',
+            'stnfs'])
+
+        if vfstype in specialFSes:
+            match_on = ['name']
+        else:
+            match_on = ['dev']
+    else:
+        match_on = [match_on]
+
+    # generate entry and criteria objects, handle invalid keys in match_on
+    entry_ip = _filesystems_entry.from_line(entry_args, kwargs)
+    try:
+        criteria = entry_ip.pick(match_on)
+
+    except KeyError:
+        filterFn = lambda key: key not in _filesystems_entry.compatibility_keys
+        invalid_keys = filter(filterFn, match_on)
+
+        msg = 'Unrecognized keys in match_on: "{0}"'.format(invalid_keys)
+        raise CommandExecutionError(msg)
+
+    # parse file, use ret to cache status
+    if not os.path.isfile(config):
+        raise CommandExecutionError('Bad config file "{0}"'.format(config))
+
+    # read in block of filesystem, block starts with '/' till empty line
+    try:
+        fsys_filedict = _filesystems(config, False)
+        for fsys_view in six.viewitems(fsys_filedict):
+            if criteria.match(fsys_view):
+                ret = 'present'
+                if entry_ip.match(fsys_view):
+                    view_lines.append(fsys_view)
+                else:
+                    ret = 'change'
+                    kv = entry_ip['name']
+                    view_lines.append((kv, entry_ip))
+            else:
+                view_lines.append(fsys_view)
+
+    except (IOError, OSError) as exc:
+        msg = 'Couldn\'t read from {0}: {1}'
+        raise CommandExecutionError(msg.format(config, exc))
+
+    # add line if not present or changed
+    if ret is None:
+        for dict_view in six.viewitems(entry_ip.dict_from_entry()):
+            view_lines.append(dict_view)
+        ret = 'new'
+
+    if ret != 'present':  # ret in ['new', 'change']:
+        try:
+            with salt.utils.files.fopen(config, 'wb') as ofile:
+                # The line was changed, commit it!
+                for fsys_view in view_lines:
+                    entry = fsys_view[1]
+                    mystrg = _filesystems_entry.dict_to_lines(entry)
+                    ofile.writelines(salt.utils.data.encode(mystrg))
+        except (IOError, OSError):
+            msg = 'File not writable {0}'
+            raise CommandExecutionError(msg.format(config))
```

Comment:
We don't need to assign this to a variable if all we're doing is raising an exception with it.

Reply:
Due to cut & paste, as similar usage throughout existing file.

Reply:
done